src/function-builder.ts

```diff
@@ -44,45 +45,79 @@ import * as storage from './providers/storage';
 /**
  * Assert that the runtime options passed in are valid.
  * @param runtimeOptions object containing memory and timeout information.
- * @throws { Error } Memory and TimeoutSeconds values must be valid.
+ * @throws { Error } FailurePolicy, Memory and TimeoutSeconds values must be
+ *     valid.
  */
-function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
-  if (
-    runtimeOptions.memory &&
-    !_.includes(VALID_MEMORY_OPTIONS, runtimeOptions.memory)
-  ) {
-    throw new Error(
-      `The only valid memory allocation values are: ${VALID_MEMORY_OPTIONS.join(
-        ', '
-      )}`
-    );
+function assertRuntimeOptionsValidity(runtimeOptions: RuntimeOptions): void {
+  if (_.isObjectLike(runtimeOptions) === false) {
+    throw new Error('RuntimeOptions must be an object.');
   }
-  if (
-    runtimeOptions.timeoutSeconds > MAX_TIMEOUT_SECONDS ||
-    runtimeOptions.timeoutSeconds < 0
-  ) {
-    throw new Error(
-      `TimeoutSeconds must be between 0 and ${MAX_TIMEOUT_SECONDS}`
-    );
+
+  const { failurePolicy, memory, timeoutSeconds } = runtimeOptions;
```

Comment:
I'm not familiar with this syntax - is this a shorthand for something like this `const failurePolicy = runtimeOptions.failurePolicy`?

Reply:
This is an ES destructing assignment.

Specification: https://www.ecma-international.org/ecma-262/9.0/index.html#sec-destructuring-assignment

MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

Reply:
👍